extensions/sql-assessment/src/assessmentResultGrid.ts

```diff
@@ -121,9 +121,9 @@ export class AssessmentResultGrid implements vscode.Disposable {
 
 	public async displayResult(asmtResult: azdata.SqlAssessmentResult, method: AssessmentType) {
 		this.asmtType = method;
-		this.dataItems = asmtResult.items;
+		this.dataItems = this.filterOutNotSupportedKind(asmtResult.items);
 		await this.table.updateProperties({
-			'data': asmtResult.items.map(item => this.convertToDataView(item))
+			'data': this.filterOutNotSupportedKind(asmtResult.items).map(item => this.convertToDataView(item))
```

Comment:
Instead of filtering twice just store the filtered results in a separate var and use it for both.